### ISSUE TITLE
Repeated Operations I Lab

### ISSUE THREAD

**Issue by mbcann01** [brad-cannell/intro.r.public#34]
# Overview

From [ClickUp](https://app.clickup.com/t/td2th2): When I first made this lab, I did so in a bit of a hurry. I think it has room for improvement. 

## Development Notes

- **If nothing else** turn question 2 on the current lab into a multiple choice question or just get rid of it. ("Tasks 4 through 7 above essentially required you to do the same thing in four different slightly different ways. For this particular task, which method did you prefer, and why?")
- We should just read-in Rds files that are ready for the repeated operations examples. Nothing extraneous. (actually, maybe do the "extraneous" stuff in the lab, save that data, and then call it up again for the module quiz?
- This lab should cover much of what's in the R4Epi chapters fairly closely.
  - [Introduction](https://www.r4epi.com/introduction-to-repeated-operations.html)
  - [Writing Functions](https://www.r4epi.com/writing-functions.html)
  - [Column-wise operations](https://www.r4epi.com/column-wise-operations-in-dplyr.html)

### Set up possibilities

- [x] Import the class survey data
- [ ] Import the temps 2012 data

### Writing Functions ideas

- [ ] When to write functions. Find an example from one of the other labs/quizzes where we repeated code more than 2. Then ask if this is a good time to write function.
- [ ] Parts of a function
- [ ] How to write fucntions
- [ ] Giving functions default values
- [ ] Return values
- [ ] Lexical scoping
- [ ] Tidy evaluation
- [ ] Write a function to convert 9's to missing (already in the lab)
- [ ] Write a function to calculate continuous stats using the temps data (https://app.clickup.com/t/td2th7)
- [ ] Write a function to calculate categorical stats using class survey data (https://app.clickup.com/t/td2thb)

### Column-wise operations ideas

- [x] Rename multiple columns with rename_with and a character vector
- [ ] The across() function arguments 
  - [ ] Assign new variable names with across() inside mutate() using the .names argument.
- [ ] Across() with mutate(). 
  - [ ] Convert multiple column types at once with across().
  - [ ] Convert character vectors to factors using across()
- [ ] Across() with summarise()
  - [ ] Calculate the number of missing for each variable using across() with summarize. 
  - [ ] Pivot summary stats table.
- [ ] Use across() with filter() to do complete case analysis.
- [ ]  In the restructuring data lab, I wanted to graph mean repeat time by initial incident complaint. This was going to be really hard without repeated operations. Show them how to do it with repeated operations. (https://app.clickup.com/t/td2th9)

# Tasks

- [ ] Turn question 2 on the current lab into a multiple choice question or just eliminate it. ("Tasks 4 through 7 above essentially required you to do the same thing in four different slightly different ways. For this particular task, which method did you prefer, and why?")
- [ ] Update the lab in Canvas
- [ ] Consider using one data set. Keep the same basic content but use one (simulated) data set instead of importing multiple data sets. Perhaps even have the data simulation code at the top of the file.
- [ ] Use the lab as a template for the module quiz (#19)

**Comment by mbcann01** [comment]
2023-07-23: I didn't get this finished in time to add it to Canvas this semester. Try again next semester.
- I was working on this lab and the quiz (#19) in tandem. 
- I reviewed task 1-3 in the lab.
- I made some minor wording and formatting updates to `lab_repeated_operations_I.Rmd`; however, those changes aren't reflected in the text in Canvas or the Google Doc.
- When you start again, finish reviewing task 4 in the lab and then make a similar task in the module quiz. You started to do this, but didn't finish.
